packages/utils/src/normalize.ts

```diff
@@ -79,7 +79,11 @@ function visit(
   const [memoize, unmemoize] = memo;
 
   // Get the simple cases out of the way first
-  if (value === null || (['number', 'boolean', 'string'].includes(typeof value) && !isNaN(value))) {
+  if (
+    value === undefined ||
+    value === null ||
```

Comment:
super-l (just with bundle size in mind but feel free to ignore):
```suggestion
    value == null ||
```

This should be identical, just a little shorter, right?

Reply:
You're right. I even thought about this but I stopped micro-optimizing for bundle because I assume terser will optimize this stuff away anyhow. Here I am still gonna go with your change because I don't know if it catches ths.